Doc/library/reprlib.rst

```diff
@@ -123,6 +123,62 @@ which format specific object types.
    similar manner as :attr:`maxstring`.  The default is ``20``.
 
 
+.. attribute:: Repr.indent
+
+   If this attribute is set to ``None`` (the default), the output is formatted
+   with no line breaks or indentation, like the standard :func:`repr`.
```

Comment:
I think a more sensible default value would be just `""` (an empty string). In other words, *only support* the string indentation variant. This will let users say `4 * " "` if they want for spaces, and `"\t"` if they want tabs. The implementation becomes simpler then too as you only have to deal with strings.

In this case we wouldn't be handling the case where you explode arguments but don't indent. I think this is fair game as the attribute is called `indent` anyway.

Reply:
Thanks for the suggestion. The original idea was to have `None` and integer values as the the basic types for the `indent` argument because it seems appropriate to keep the basics of the `indent` attribute similar to the `indent` arguments of [`json.dump`](https://docs.python.org/3/library/json.html#json.dump) as well as [`pprint.PrettyPrinter`](https://docs.python.org/3/library/pprint.html#pprint.PrettyPrinter) which use `None` and integer values, respectively. On top of that, allowing string values is a useful extension for certain cases. (`"\t"` being an obvious one.) Taking all this into consideration, I don't feel like having `reprlib.Repr.indent` support only string values would be the right choice.

Reply:
Thanks for the suggestion. The original idea was to have `None` and integer values as the the basic types for `Repr.indent` because it seems appropriate to keep the basics of the `indent` attribute similar to the `indent` arguments of [`json.dump`](https://docs.python.org/3/library/json.html#json.dump) as well as [`pprint.PrettyPrinter`](https://docs.python.org/3/library/pprint.html#pprint.PrettyPrinter) which use `None` and integer values, respectively. On top of that, allowing string values is a useful extension for certain cases. (`"\t"` being an obvious one.) Taking this into consideration, I think I don't feel like having `reprlib.Repr.indent` support only string values would be the right choice, but could lead to confusion.

Reply:
While such an approach is seemingly alluring, upon close inspection this doesn't appear to actually allow simplifying the code much, as a `""` value would is still just as much a special case that would need to be handled specially, just as we do now (unless we change the default to explode the repr, which is almost certainly not what we want)—just with less clarity, since `None` more obviously is a special default case than an empty string. The only block that could be obviously eliminated is 

```python
        if isinstance(indent, int):
            if indent < 0:
                raise ValueError(
                    f'Repr.indent cannot be negative int (was {indent!r})'
                )
            indent *= ' '
```

which is by a fair margin most straightforward and easy to understand code in the added function, relative to the much more complex string-munging bits, and that is only due to not accepting `int` arguments; changing the default value would not enable any further simplification.

By contrast, the behavior is potentially more confusing and less obvious and useful to users, since `""` is magically handled differently than any other string and breaks the natural expectation arising from the pattern when passing `"    "`, `"  "`, `" "` or `" " * 4`, `" " * 2`, `" " * 1`, and that leaves no way to set an indent level of zero but still explode a string—I'm not sure how else this would be done.

Given all that, as well as the points raised by @finefoot above and the fact that implementation is complete and has fairly extensive unit and manual tests for a wide array of cases, I'd suggest retaining the design as-is. If supporting integer as well as string arguments is a hard blocker, then at worst that support could be removed while retaining `None` as a default value, allowing `""` to be passed, as changing that would not make the implementation any more simple while artificially limiting functionality and quite possibly confusing users).